.github/workflows/release.yml

```diff
@@ -0,0 +1,101 @@
+name: Release
+
+on:
+  push:
+    branches:
+      - main
+
+# This workflow will run changesets depending on two different scenarios:
+#
+# 1. If we are landing a specific commit into main (Author PR), then
+#    changesets will check if there are changes verifying the Markdown files
+#    generated automatically:
+#
+#    a) There are new versions and there's NO Release PR, then the changesets
+#    action will create a new Release PR.
+#
+#    b) There's a Release PR, then the changesets action will update the
+#    existing Release PR with the new commit.
+#
+#    NOTE: (in both cases, changesets will modify the new version in
+#    package.json for each package, and will remove the MD files as part of the
+#    Release PR).
+#
+# 2. If we are landing the Release PR into main, then the changesets action
+#    will publish the changes to npm.
+#
+# For more info about this workflow, see:
+# https://github.com/changesets/action#usage
+jobs:
+  release:
+    name: Release
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Repo
+        uses: actions/checkout@master
+        with:
+          # This makes Actions fetch all Git history so that Changesets can
+          # generate changelogs with the correct commits
+          fetch-depth: 0
+
+      - name: Setup Node.js 16.x
+        uses: actions/setup-node@master
+        with:
+          node-version: 16.x
```

Comment:
Ultimately, this is what indicates what version we target. If we build under 16, we target 16 as our babel/rollup config just targets whatever version of Node it runs under. If we wanted to target 12, we'd need to change this (and probably add back in testing to verify it works on 12).

Reply:
I don't see a reason why we should break things for older versions - we just need to make sure that things work ok on newer versions! So I think it's fine to have testing for 12.x + 16.x, but only do the release for 12.x.

Reply:
Yeah, OK. Though I really wanted some of those new Node features :D